### PR TITLE
Support changed Guava SimpleTimeLimiter API

### PR DESCRIPTION
**Pull request by lammel**
During the upgrade to Graylog v2.4.0 we had issues with bringing up the server.
The reason was an error DnsResolver caused an error:

```
2018-01-05T09:00:06.128+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Error injecting constructor, java.lang.IllegalAccessError: tried to access method com.google.common.util.concurrent.SimpleTimeLimiter.<init>(Ljava/util/concurrent/ExecutorService;)V from class org.graylog.plugin.filter.dns.DnsResolverFilter
```

This pull request changes the SimpleTimeLimiter to use the new Guava API.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
     <groupId>org.graylog.plugins</groupId>
     <artifactId>graylog-plugin-dnsresolver</artifactId>
-    <version>1.2.0-SNAPSHOT</version>
+    <version>1.1.3-nt1</version>
```
Comment (joschi) [inline pom.xml]: Please remove the custom version and use the old version string.

File: pom.xml [hunk 2]
```diff
@@ -38,6 +38,11 @@
             <version>${graylog2.version}</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>23.6-jre</version>
+        </dependency>
```
Comment (joschi) [inline pom.xml]: Please migrate the POM to use the official `graylog-plugin-parent` parent POM.
This way you don't have to manually add any dependencies which are pulled in by Graylog itself.

Example: https://github.com/graylog-labs/graylog-plugin-internal-logs/blob/2.4.0-beta.2/pom.xml

File: pom.xml (hunk 1 above)
Comment (joschi) [reply pom.xml]: Also please update the `graylog2.version` property.

The changed plugin will not run with Graylog 2.0.0.

File: src/main/java/org/graylog/plugin/filter/dns/DnsResolverFilterMetaData.java [hunk 3]
```diff
@@ -49,7 +49,7 @@ public URI getURL() {
 
     @Override
     public Version getVersion() {
-        return new Version(1, 2, 0, "SNAPSHOT");
+        return new Version(1, 1, 3, "nt1");
```
Comment (joschi) [inline src/main/java/org/graylog/plugin/filter/dns/DnsResolverFilterMetaData.java]: Please fetch the version number from a properties file generated by the Maven build.

Example:
* https://github.com/graylog-labs/graylog-plugin-internal-logs/blob/2.4.0-beta.2/src/main/java/org/graylog/plugins/internallogs/InternalLogsInputPluginMetaData.java
* https://github.com/graylog-labs/graylog-plugin-internal-logs/blob/2.4.0-beta.2/src/main/resources/org.graylog.plugins.graylog-plugin-internal-logs/graylog-plugin.properties
* https://github.com/graylog-labs/graylog-plugin-internal-logs/blob/2.4.0-beta.2/pom.xml#L91-L101

**Review by joschi — CHANGES_REQUESTED** [review]
Thanks for your contribution to this plugin!

There are a few things we have to change before we can merge this PR.

Also, please sign the CLA so we can move on with this PR.

File: pom.xml (hunk 1 above)
Comment (lammel) [reply pom.xml]: So I will update the graylog2.version to 2.4.0, so release 1.2.x will target graylog 2.4+ then.

**Comment by lammel** [comment]
Updated based on feedback and synced handling for version with graylog-plugin-internal-logs plugin.

Should be ready to merge. Tested on our Graylog 2.4 server.

**Review by joschi — CHANGES_REQUESTED** [review]
Please remove the files (`.classpath`, `.project`) and directories ( `.settings/`, `.vscode/`) specific to Eclipse IDE and VS Code.
They probably not have been committed. 😉 

Otherwise the PR looks good to me!

**Comment by lammel** [comment]
Argh. Should have used the command line ;-)
Removed again, maybe some of that stuff should be added to .gitignore

Happy merging.

**Comment by joschi** [comment]
> maybe some of that stuff should be added to .gitignore

FWIW, you can add generic IDE files to your global `.gitignore` file:
https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

**Review by joschi — APPROVED** [review]

**Comment by joschi** [comment]
@lammel Thanks!

**Pull request merged**